src/shared/extension-release.c

```diff
@@ -119,14 +119,18 @@ int extension_release_validate(
         return 1;
 }
 
-int parse_env_extension_hierarchies(char ***ret_hierarchies) {
+int parse_env_extension_hierarchies(char ***ret_hierarchies, char const *env_hierarchy) {
         _cleanup_free_ char **l = NULL;
         int r;
 
```

Comment:
either add an `assert(env_hierarchy)` here, because the argument must be passed. or add:

```c
if (!env_hierarchy)
        env_hierarchy = "SYSTEMD_SYSEXT_HIERARCHIES";
```

i.e. treat NULL as the status quo ante